### PR TITLE
.editorconfig

### LINKED ISSUE #355: Add .editorconfig to this repository
So that editors will automatically indent with 4 spaces and the like.

### PR DESCRIPTION
**Pull request by littlewhitecloud**
Improve #355

### CODE REVIEW

**Comment by littlewhitecloud** [comment]
lf end
4 space
utf-8

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[*]
+indent_style = space
+end_of_line = lf
```
Comment (Akuli) [inline .editorconfig:3]: Git for Windows converts the files to CRLF automagically, so even though they are committed as LF, cloning the repository on Windows will actually give you CRLF.

```suggestion
```

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+charset = utf-8
+trim_trailing_whitespace = true
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (Akuli) [inline .editorconfig:9]: We don't need markdown to be special, but we do need to indent Makefiles with tabs. There's also one test file that is indented with a tab.

```suggestion
[{Makefile,Makefile.*,tests/syntax_error/tab.jou}]
indent_style = tab
```

**Comment by Akuli** [comment]
Maybe also add to github actions some tool that checks whether files follow the editorconfig? It could go to `.github/workflows/linux.yml`. Searching "editorconfig validator" gives me https://github.com/editorconfig-checker/editorconfig-checker which seems useful.

**Comment by littlewhitecloud** [comment]
I agree that.But I am not good at adding GitHub actions(lol)

**Comment by Akuli** [comment]
We could copy/paste the example in https://github.com/editorconfig-checker/action-editorconfig-checker

**Comment by littlewhitecloud** [comment]
let me try later

**Comment by littlewhitecloud** [comment]
Oh no, too many errors 
I think we should add something like autofix

**Comment by littlewhitecloud** [comment]
image
That's really strange.
typecheck.jou only has 1147 lines, but it say 1176 has trailing whitespace

**Comment by Akuli** [comment]
The line numbers seem to be off for whatever reason, but I searched for spaces at ends of the lines and fixed the remaining ones.

**Comment by littlewhitecloud** [comment]
Finally…

**Review by Akuli — APPROVED** [review]
Thanks for working on this!

**Pull request merged**